webapi/src/main/scala/org/knora/webapi/responders/admin/UsersResponderADM.scala

```diff
@@ -261,28 +261,19 @@ class UsersResponderADM(responderData: ResponderData) extends Responder(responde
 
             // check if we want to change the email
             _ = if (changeUserRequest.email.isDefined) {
-                currentUserInformation.map { user =>
-                    // check if current email differs from the one in the change request
-                    if (!user.email.equals(changeUserRequest.email.get)) {
-                        // check if new email is free
-                        userByEmailExists(changeUserRequest.email.get).map(result =>
-                            if (result) throw DuplicateValueException("A user with this email exists already. Cannot change")
-                        )
-                    }
+                val email = currentUserInformation.get.email
+                if (!email.equals(changeUserRequest.email.get)) {
+                    val result = userByEmailExists(changeUserRequest.email.get)
```

Comment:
Ups, using Await is unfortunately a „no go“. Await is blocking and thus evil.

Reply:
https://doc.akka.io/docs/akka/2.5/futures.html#use-with-actors

Reply:
> Warning

> Await.result and Await.ready are provided for exceptional situations where you must block, a good rule of thumb is to only use them if you know why you must block. For all other cases, use asynchronous composition as described below.

Reply:
I see, but why would we want concurrent execution of the query, as long as we can't guarantee the uniqueness of the values? Otherwise we would have to rollback, as soon as the results are in...

Reply:
The idea is to free up CPU threads so they can do something else (e.g. process a request from another user) while waiting for a result from the triplestore.